src/core/lib/iomgr/ev_poll_posix.cc

```diff
@@ -557,6 +557,23 @@ static void fd_notify_on_error(grpc_fd* fd, grpc_closure* closure) {
   abort();
 }
 
+static void fd_set_readable(grpc_fd* fd) {
+  gpr_mu_lock(&fd->mu);
+  set_ready_locked(fd, &fd->read_closure);
+  gpr_mu_unlock(&fd->mu);
+}
+
+static void fd_set_writable(grpc_fd* fd) {
+  gpr_mu_lock(&fd->mu);
+  set_ready_locked(fd, &fd->write_closure);
+  gpr_mu_unlock(&fd->mu);
+}
+
+static void fd_set_error(grpc_fd* fd) {
+  gpr_log(GPR_ERROR, "Polling engine does not support tracking errors.");
+  abort();
```

Comment:
abort() seems a little harsh here. Can we fail silently and log an error message?

Reply:
We can but it won't be symmetric with fd_notify_on_error anymore

Reply:
What do you think?

Reply:
Actually on a second thought, calling abort() even at `fd_notify_on_error` seems a bit harsh. If we want error tracking but are using `poll` polling engine, we should log an error and still continue instead of aborting.

Reply:
I guess the question is: Is it really a fatal-error (i.e grpc cannot continue since we will have totally unexpected/potentially-harmful behavior) to call `fd_notify_on_error` and `fd_set_error` if the underlying polling engine is 'poll'. If so, I am ok with abort(). If not, we should just log an error message..

Reply:
It's more like a programmer error

Reply:
similar to 
https://github.com/grpc/grpc/blob/7fc47f27d5956fdc8ca7a3567ea8313e7fe742b5/src/core/lib/iomgr/ev_poll_posix.cc#L494

Reply:
The case you mentioned would abort() for all polling engines. Not just poll.. So that is my concern.

How is the caller ensuring that this should not be called for poll polling engine?
(for example, we if we found a bug in epoll-based polling engine and asked customers to switch to poll polling engine for time being, then are we sure that fd_set_error/fd_notify_on_error won't abort? )

Reply:
Through can_track_err field. The idea is that the caller should make sure that the polling engine can track errors before calling any of these. But, I also like your approach of soft errors. I could simply call the closure with GRPC_ERROR_CANCELLED :)

Reply:
Done

Reply:
Removing abort from grpc_fd_notify_on_error in #16166